chore(app_agent_receiver): add new ExceptionContext  property (#3911)
grafana/agent: CHANGELOG.md

- Use Go 1.20.4 for builds. (@tpaschalis)

> - Integrate the new ExceptionContext which recently added to the Faro Web-SDD in the
>   app_agent_receiver Payload. (@codecapitano)

Reviewer: ```suggestion
- Integrate the new Faro Web-SDK ExceptionContext with the app_agent_receiver Payload. (@codecapitano)
```
(Faro-SDD -> Faro-SDK?)
(which recently added -> which [was] recently added)

  compiler issues are resolved and 32-bit ARM builds are stable. (@rfratto)

- Agent Management: `agent_management.api_url` config field has been replaced by
  `agent_management.host`. The API path and version is now defined by the Agent. (@jcreixell)

- Agent Management: `agent_management.protocol` config field now allows defining "http" and "https" explicitly. Previously, "http" was previously used for both, with the actual protocol used inferred from the api url, which led to confusion. When upgrading, make sure to set to "https" when replacing `api_url` with `host`. (@jcreixell)

- Agent Management: `agent_management.remote_config_cache_location` config field has been replaced by

>   `agent_management.remote_configuration.cache_location`. (@jcreixell)

Reviewer: Q: are these formatting changes intentional ? (just asking in case they're not)

Author: Oh no they are not. This is cause by the auto formatter of my editor.
